test/Jacobian_test.cpp

```diff
@@ -0,0 +1,394 @@
+/*******************************************************************************
+
+   Tests for jacobian computation methods.
+
+   This file is part of XAD, a comprehensive C++ library for
+   automatic differentiation.
+
+   Copyright (C) 2010-2024 Xcelerit Computing Ltd.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Affero General Public License as published
+   by the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+******************************************************************************/
+
+#define _USE_MATH_DEFINES
+#include <XAD/Jacobian.hpp>
+#include <XAD/XAD.hpp>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <functional>
+#include <list>
+#include <numeric>
+#include <vector>
+
+TEST(JacobianTest, SimpleAdjoint)
+{
+    typedef xad::adj<double> mode;
+    typedef mode::tape_type tape_type;
+    typedef mode::active_type AD;
+
+    tape_type tape;
+
+    std::vector<AD> x = {3.0, 1.0};
```

Comment:
I have no idea why Codacy complains about these supposedly unused variables - they clearly are used. We're going to add them to the ignore list later, so don't worry about them.

Reply:
Thank you, Codacy also seems to have some issues with my doc specifically the main headers? If you could fix that too that'd be great.